napari/layers/points/points.py

```diff
@@ -408,13 +420,13 @@ def selected_data(self, selected_data):
 
         # Update properties based on selected points
         index = self._selected_data
-        edge_colors = list(set([self.edge_colors[i] for i in index]))
+        edge_colors = ColorArray(np.unique(self.edge_colors.rgba, axis=0))
```

Comment:
This logic is wrong here - it is missing the `index` - should be something more like

```python
        edge_colors = ColorArray(np.unique(self.edge_colors[index].rgba, axis=0))
```